From b7358bcebef69ce2616b2fa753ddcce216e09c0c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:14 -0400
Subject: [tools] perf/report: Add report.children config option

Message-id: <1410181752-25631-250-git-send-email-jolsa@redhat.com>
Patchwork-id: 91895
O-Subject: [PATCH RHEL7.1 BZ1134356 249/307] perf report: Add report.children config option
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 8d8e645ceafd726b8317949f899e4b3acfb20d29
Author: Namhyung Kim <namhyung@kernel.org>
Date: Tue Jan 22 18:09:46 2013 +0900

description
===========
Add report.children config option for setting default value of
callchain accumulation.  It affects the report output only if
perf.data contains callchain info.

A user can write .perfconfig file like below to enable accumulation
by default:

  $ cat ~/.perfconfig
  [report]
  children = true

And it can be disabled through command line:

  $ perf report --no-children
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index f27a8aa..6cac509 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -72,6 +72,10 @@ static int report__config(const char *var, const char *value, void *cb)
   rep->min_percent = strtof(value, NULL);
   return 0;
  }
+ if (!strcmp(var, "report.children")) {
+  symbol_conf.cumulate_callchain = perf_config_bool(var, value);
+  return 0;
+ }
 
  return perf_default_config(var, value, cb);
 }
-- 
1.7.1